src/cpuid.cpp

```diff
@@ -55,8 +55,21 @@ POSSIBILITY OF SUCH DAMAGE.
 #endif
 
 
-#if TORRENT_HAS_ARM && TORRENT_HAS_AUXV
+#if defined(TORRENT_ANDROID) && TORRENT_HAS_AUXV
+#include <dlfcn.h>
+namespace {
+unsigned long int _getauxval(unsigned long int type)
+{
+    using getauxval_t = unsigned long int(*)(unsigned long int);
+    getauxval_t p_getauxval = reinterpret_cast<getauxval_t>(dlsym(RTLD_DEFAULT, "getauxval"));
+    if (p_getauxval == nullptr)
+        return 0;
+    return p_getauxval(type);
+}
+}
+#elif TORRENT_HAS_ARM && TORRENT_HAS_AUXV
 #include <sys/auxv.h>
+#define _getauxval getauxval
```

Comment:
now, if you introduce a define, you're definitely in trouble using a reserved word. I think it's probably best to use a name that doesn't start with an underscore.

Reply:
in fact, even in the anonymous namespace, it would overload with the global namespace, so it seems questionable even without the define

Reply:
You're definitely right, I will rename it ``helper_getauxval``.

Reply:
fixed